vars/golang.groovy

```diff
@@ -31,13 +33,20 @@ def call(opts = []) {
               def originalWs = "${WORKSPACE}"
               ws("${originalWs}\\src\\github.com\\${jobName}") {
                 def goEnv = ["GOROOT=${root}", "GOPATH=${originalWs}", "PATH=$PATH;${root}\\bin;${originalWs}\\bin"]
+                if (dep != "gx") {
+                  goEnv.add("GO111MODULE=on")
+                }
                 withEnv(goEnv + env) {
                   checkout scm
-                  bat 'go get -v github.com/whyrusleeping/gx'
-                  bat 'go get -v github.com/whyrusleeping/gx-go'
-                  bat 'go get -v github.com/jstemmer/go-junit-report'
-                  bat 'gx --verbose install --global'
-                  bat 'gx-go rewrite'
+                  if (dep == "gx") {
+                    bat 'go get -v github.com/whyrusleeping/gx'
+                    bat 'go get -v github.com/whyrusleeping/gx-go'
+                    bat 'go get -v github.com/jstemmer/go-junit-report'
+                    bat 'gx --verbose install --global'
+                    bat 'gx-go rewrite'
+                  } else {
+                    bat 'go mod download'
```

Comment:
nit: our gomod path triggers whenever the user provides a value for `dep` and iff the value is anything other than `gx`, not necessarily `go-mod` as the README.md suggests.